tests/test_custom_ai.py

```
@@ -0,0 +1,153 @@
+
+
+@pytest.mark.parametrize(
+    "controls",
+    [
+        launch_controls(gas=math.nan),
+        launch_controls(gas=1.1),
+        launch_controls(steer=-1.2),
+    ],
+)
+def test_pack_controls_rejects_nonfinite_or_out_of_range(controls):
+    with pytest.raises(ValueError):
```

Ruff (0.15.17)

[warning] `pytest.raises(ValueError)` is too broad, set the `match` parameter or use a more specific exception

(PT011)